### PR TITLE
Replace GLEW with libepoxy

### PR DESCRIPTION
**Pull request by linkmauve**
This removes the dependency on X11, and allows for a much more future-proof OpenGL context, with core, GLES, EGL and everything supported, see https://github.com/anholt/libepoxy#why-not-use-libglew

Also drops the cmake version required back to 2.8.8.

### CODE REVIEW

File: cpp/crossplatform/opengl.h [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-#ifndef OPENAGE_CROSSPLATFORM_OPENGL_H_
-#define OPENAGE_CROSSPLATFORM_OPENGL_H_
-
-#include <GL/glew.h>
-
-#ifdef __APPLE__
-#  include <OpenGL/gl.h>
-#else
-#  include <GL/gl.h>
-#endif
```
Comment (TheJJ) [inline cpp/crossplatform/opengl.h]: can anyone try or confirm that this'll still work on osx systems?

**Comment by linkmauve** [comment]
For the record, the last missing thing for this PR is a PPA containing libepoxy for Ubuntu 12.04.

**Comment by mic-e** [comment]
oooh, Wayland support :D

Congratulations on the round issue number!

Travis just _got_ to upgrade their build bots to Ubuntu 14.04 at some point... after all, Ubuntu 12.04 will reach EOL in about a month...

**Comment by TheJJ** [comment]
nah, LTS versions got 5 years, two more to go..
i'll see if i can package `libepoxy` in our ppa.

**Comment by mic-e** [comment]
5 years support is only for Ubuntu Server LTS. The client stuff has 3 years.

**Comment by franciscod** [comment]
holy cow, this looks awesome!!

**Comment by mic-e** [comment]
`libepoxy` currently has an [unmerged pull request](https://github.com/anholt/libepoxy/pull/45) that would supposedly allow cross-compiling to Windows.

**Comment by janisozaur** [comment]
(Unfortunately) Travis are using server flavour, so there's no pressure on them so far from this side.
http://docs.travis-ci.com/user/ci-environment/#CI-environment-OS

**Comment by TheJJ** [comment]
bäm :+1:

**Comment by mic-e** [comment]
wheeee (has any OSX user actually tested this before it was merged, or is it their fault for not reacting to our pleas?)

**Pull request merged**